### DISCUSSION TITLE
Is `lint-staged` still relevant when using Turborepo?

### DISCUSSION THREAD

**Discussion by jplwood** [vercel/turborepo#2595] [Help] [answered]
I recently set up a new monorepo for my team using turborepo (first-time user). In most of our repositories, we have the usual build tooling (tsc building, eslint linting, jest testing), but we also usually set up [`husky`](https://typicode.github.io/husky/#/) for git hooks and pair it with [`lint-staged`](https://github.com/okonet/lint-staged). I had no problem setting up husky+lint-staged, but the usual configuration for it got me thinking: 

Is lint-staged still relevant when using turborepo?

To demonstrate, my package.json has this `lint-staged` config:

```json
  "lint-staged": {
    "apps/**/*.{js,ts,jsx,tsx}": [
      "eslint --fix"
    ]
  }
```

This works just fine, but it clearly has one problem, it's not using my `"lint": "turbo run lint"` script and leveraging my cache. And if I _do_ use my turbo script here, things get a little awkward because lint-staged is running the eslint command against staged files, while turbo has its own agenda when figuring out which files to target. Should `husky` simply run my `turbo run lint` script instead of using `lint-staged`? Can you apply [`turbo run --filter`s](https://turbo.build/repo/docs/core-concepts/monorepos/filtering#filter-by-changed-workspaces) only to staged files? (the docs aren't exactly exhaustive/clear about the applications for git-history-related filter possibilities)

I'm still pretty new to Turborepo, and I haven't fully explored this, but I couldn't find any blogs/docs about this tool in particular as it relates to Turbo, so wanted to raise the question to see if anyone else has considered it.

**Comment by nathanhammond** [comment, marked as answer]
This is actually a fantastic question!

In this scenario Turborepo doesn't provide any particularly compelling benefits:
- You already have an enumerated list of things you want to lint.
- You'll _literally never_ see those things again since it's `--fix` on a commit hook.

In theory, the incremental cache from ESLint would imply that you do not need to explicitly specify the staged files in order to get approximately equivalent performance to what Turborepo can provide, and, since it's regarding local changes, it won't ever provide value for others via shared caching.

Now, if your `lint` task has dependencies, (e.g. type generation) you can still leverage Turborepo to invoke the command using an argument separator and passing the list of files behind an argument separator `--` into ESLint from there. But the conditions where this is useful are limited to situations where your `lint` task has dependencies.

**Reply by robert-melonn** [reply]
what if lint command' dependencies are like :` "^lint"`. This not gonna work out
